rs_server/CADIP/models/cadu_download_status.py

```diff
@@ -0,0 +1,160 @@
+
+    __tablename__ = "cadu_download_status"
+
+    db_id = Column(Integer, primary_key=True, index=True, nullable=True)
+    cadu_id = Column(String, unique=True, index=True)
+    name = Column(String, unique=True, index=True)
+    available_at_station = Column(DateTime)
+    download_start = Column(DateTime)
+    download_stop = Column(DateTime)
+    status: EDownloadStatus = Column(Enum(EDownloadStatus), default=EDownloadStatus.NOT_STARTED)
+    status_fail_message = Column(String)
+
+    def __init__(self, *args, **kwargs):
+        """Invoked when creating a new record in the database table."""
+        super().__init__(*args, **kwargs)
+        self.lock = Lock()
+
+    @orm.reconstructor
+    def init_on_load(self):
+        """Invoked when retrieving an existing record from the database table."""
+        self.lock = Lock()
+
+    def not_started(self, db: Session):
+        """Update database entry to not started."""
+        with self.lock:
+            self.status = EDownloadStatus.NOT_STARTED
+            self.download_start = None
+            self.download_stop = None
+            self.status_fail_message = None
+            db.commit()
+            db.refresh(self)
+
+    def in_progress(self, db: Session, download_start: datetime = None):
+        """Update database entry to progress."""
+        with self.lock:
+            self.status = EDownloadStatus.IN_PROGRESS
+            self.download_start = download_start or datetime.now()
+            self.download_stop = None
+            self.status_fail_message = None
+            db.commit()
+            db.refresh(self)
+
+    def failed(self, db: Session, status_fail_message: str, download_stop: datetime = None):
+        """Update database entry to failed."""
+        with self.lock:
+            self.status = EDownloadStatus.FAILED
+            self.download_stop = download_stop or datetime.now()
+            self.status_fail_message = status_fail_message
+            db.commit()
+            db.refresh(self)
+
+    def done(self, db: Session, download_stop: datetime = None):
+        """Update database entry to done."""
+        with self.lock:
+            self.status = EDownloadStatus.DONE
+            self.download_stop = download_stop or datetime.now()
+            self.status_fail_message = None
+            db.commit()
+            db.refresh(self)
+
+    #######################
+    # DATABASE OPERATIONS #
+    #######################
+
+    @classmethod
+    def get_all(cls, db: Session) -> list[CaduDownloadStatus]:
+        """Get all entries in database table."""
+        return db.query(cls).all()
+
+    @classmethod
+    def get(
+        cls,
+        db: Session,
+        cadu_id: str | Column[str],
+        name: str | Column[str],
+        raise_if_missing=True,
+    ) -> CaduDownloadStatus:
+        """Get single entry by CADU ID or name."""
+
+        # Check if entry exists
+        query = db.query(cls).where((cls.cadu_id == cadu_id) | (cls.name == name))
```

Comment:
Why we need both cadu_id and name as mandatory parameters? 
1. Since you are using || operator while querying the DB, name or id can be optional, in order to simplify dependency on function call.
2. If we want to keep both parameters, then we can use AND operator to check for Id AND Name.

We should choose implementation 1 or 2, not both.